deploy/Dockerfile

```diff
@@ -17,111 +17,54 @@ WORKDIR /src
 
 # This arg is passed by docker buildx & contains the target CPU architecture (e.g., amd64, arm64, etc.)
 ARG TARGETARCH
-ARG TARGETOS
 
 ENV GOARCH=$TARGETARCH
-ENV GOOS=$TARGETOS
-
 ENV CGO_ENABLED=0
 ENV GOBIN=/usr/local/bin
 
-# Get GCR credential helper
-# Until https://github.com/GoogleCloudPlatform/docker-credential-gcr/issues/128 is addressed, we need to use
-# the existing compiled versions distributed in GitHub releases
-ARG GCR_HELPER_VERSION="2.1.8"
+# Add .docker config dir
+RUN mkdir -p /kaniko/.docker
 
-# Checksums can be obtained from https://github.com/GoogleCloudPlatform/docker-credential-gcr/releases/download/v2.1.8/checksums.txt
-ARG GCR_HELPER_SHA_linux_amd64="0da9a0878d34dc8b38ae76519b7137f3179f573cf9c84fec1b81e973b565e2f9"
-ARG GCR_HELPER_SHA_linux_386="2421d3242beb239eb640daf9dbfa4582416049a92ed93a207a0a3102fde4ea11"
-ARG GCR_HELPER_SHA_linux_s390x="5d18e62c5952d43993b7ee292f87d812ccd161d4e9d932fbc1567eb5d5013632"
-ARG GCR_HELPER_SHA_linux_arm64="9d0e67c18fb3e1e048120d06edb9e54e1c11f88139ec77dbc9830e8a8b31745c"
+COPY . .
```

Comment:
Move this step before the `go install` calls so that `go.mod`, `go.sum` and `vendor` are present when `go install` is called.